src/components/Whiteboard/ElementBehaviors/Selection/DragSelect.test.tsx

```diff
@@ -0,0 +1,123 @@
+import { SvgCanvas } from '../../SvgCanvas';
+import { DragSelect } from './DragSelect';
+
+describe('<DragSelect/>', () => {
+  let activeSlide: WhiteboardSlideInstance;
+  let dragSelectStartCoords: Point | undefined;
+  let setDragSelectStartCoords: (point: Point | undefined) => void;
+  let widgetApi: MockedWidgetApi;
+  let Wrapper: ComponentType<PropsWithChildren<{}>>;
+
+  beforeEach(() => {
+    widgetApi = mockWidgetApi();
+
+    const { whiteboardManager } = mockWhiteboardManager({
+      slides: [['slide-0', [['element-0', mockEllipseElement()]]]],
+    });
+    const activeWhiteboard = whiteboardManager.getActiveWhiteboardInstance()!;
+    activeSlide = activeWhiteboard.getSlide('slide-0');
+
+    function LayoutStateExtractor() {
+      ({ dragSelectStartCoords, setDragSelectStartCoords } = useLayoutState());
+      return null;
+    }
+
+    Wrapper = ({ children }) => (
+      <LayoutStateProvider>
+        <LayoutStateExtractor />
+        <SvgCanvas viewportWidth={200} viewportHeight={200}>
+          <WhiteboardHotkeysProvider>
+            <WhiteboardTestingContextProvider
+              whiteboardManager={whiteboardManager}
+              widgetApi={widgetApi}
+            >
+              {children}
+            </WhiteboardTestingContextProvider>
+          </WhiteboardHotkeysProvider>
+        </SvgCanvas>
+      </LayoutStateProvider>
+    );
+  });
+
+  afterEach(() => {
+    widgetApi.stop();
+  });
+
+  it('should clear the drag select start coordinates on mouse up', () => {
+    render(<DragSelect />, { wrapper: Wrapper });
+    act(() => {
+      setDragSelectStartCoords({ x: 0, y: 0 });
+    });
+
+    fireEvent.mouseUp(screen.getByTestId('drag-select-layer'));
+
+    expect(dragSelectStartCoords).toBeUndefined();
+  });
+
+  it('should not render a selection if there is a mouse move but not start coordinates', () => {
+    render(<DragSelect />, { wrapper: Wrapper });
+
+    fireEvent.mouseMove(screen.getByTestId('drag-select-layer'), {
+      clientX: 50,
+      clientY: 50,
+    });
+
+    expect(screen.queryByTestId('drag-selection')).not.toBeInTheDocument();
+  });
+
+  it('should render a selection if there is a mouse move and start coordinates', () => {
+    render(<DragSelect />, { wrapper: Wrapper });
+    act(() => {
+      setDragSelectStartCoords({ x: 0, y: 0 });
+    });
+
+    fireEvent.mouseMove(screen.getByTestId('drag-select-layer'), {
+      clientX: 50,
+      clientY: 50,
+    });
+
+    expect(screen.getByTestId('drag-selection')).toBeInTheDocument();
+  });
+
+  it('should select elements if there is a mouse move', () => {
+    render(<DragSelect />, { wrapper: Wrapper });
+    act(() => {
+      setDragSelectStartCoords({ x: 0, y: 0 });
+    });
+
+    fireEvent.mouseMove(screen.getByTestId('drag-select-layer'), {
+      clientX: 50,
+      clientY: 50,
+    });
+
+    expect(activeSlide.getActiveElementIds()).toEqual(['element-0']);
```

Comment:
Would it be valuable to extend this test to a scenario where you have more than one active element? thinking of when your draggable area intersects with more than one shape / element.

Reply:
All this should hopefully be covered here https://github.com/nordeck/matrix-neoboard/pull/337/files#diff-141cbf23c431d9177f59fc54be3f68eabf0b58dea8bd24701892dc942be63890

But a quick test here can't hurt.

Reply:
I've also added an element outside of the selection.